newscript.go

```diff
@@ -0,0 +1,405 @@
+package main
+
+//include a loading bar
+//TODO: Use filePath.join
+//automated "yes"er updating the config file
+//TODO check if it is a device
+//TODO append method name to error
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"errors"
+	"log"
+	"path/filepath"
+	"os/user"
+	//"bufio"
+)
+
+var kernelVersion = "4.12.7"
+var workingDir = ""
+var linuxVersion = fmt.Sprintf("linux-%s", kernelVersion)
+var homeDir = ""
+
+
+func setup() error{
+	fmt.Printf("-------- Setting up \n")
+	dir,err := os.Getwd(); 
+	if err != nil {
+		return err
+	}
+	workingDir = dir
+	fmt.Printf("Working dir is %s\n", workingDir)
+	cmd16 := exec.Command("sudo", "apt-get", "install", "git", "golang", "build-essential", "git-core", "gitk", "git-gui", "subversion", "curl", "python2.7", "libyaml-dev", "liblzma-dev") 
+	err = cmd16.Run() 
+	if err != nil { 
+		return err
+	}
+	/*err = blankBootstick()
+	if err != nil {
+		return err
+	}*/
+	usr, err := user.Current()
+    	if err != nil {
+        	return err
+    	}
+	homeDir = usr.HomeDir
+	return nil
+
+}
+
+//User input for putting custom chrome image on bootstick
+func blankBootstick() error{
+	fmt.Printf("-------- Creating bootstick \n")
+	var imageLoc string
+	var location string 	
+	for true {
+		fmt.Printf("What image would you like to put onto your bootstick (provide location for iso file)?\n")	
+		_, err := fmt.Scanf("%s",&imageLoc)
+		if err != nil {
+			return err	
+		}
+		if _, err = os.Stat(imageLoc); err != nil{
+			fmt.Printf("Please provide a valid file name. %s has error %v", imageLoc, err)
+		} else {
+			break
+		}
+	}
+	for true {
+		fmt.Printf("Where is your bootstick (/dev/sda?)")	
+		_, err := fmt.Scanf("%s",&location)
+		if err != nil {
+			return err	
+		}
+		if _, err = os.Stat(location); err != nil{
+			fmt.Printf("Please provide a valid location name. %s has error %v", location, err)
+		} else {
+			break
+		}
+	}
+	cmd20 := exec.Command("sudo", "dd", fmt.Sprintf("if=%s", imageLoc), fmt.Sprintf("of=%s", location))
+	err := cmd20.Run()
+	if err != nil {
+		return err
+	}
+	return nil	
+}
+
+
+func cleanup() error {	
+	filesToRemove := [...]string{linuxVersion, ".git", "NiChrome", "vboot_reference"}
+	fmt.Printf("-------- Removing problematic files %v\n", filesToRemove)
+	for _,file := range filesToRemove{
+		if _, err := os.Stat(file); err != nil {
+			continue		
+		} 
+		cmd15 := exec.Command("sudo", "rm", "-r", file)
```

Comment:
why not use the go os.Remove function? using sudo in programs is worrisome. Why do you need to use sudo?

Reply:
why cmd15?

Reply:
I'll need to renumber it. Sudo was needed because kernel wasn't giving permission I think. Trying now with os.Remove.

Reply:
Also I don't see why you remove .git?